tests/components/picnic/test_config_flow.py

```diff
@@ -120,5 +131,111 @@ async def test_form_exception(hass):
             },
         )
 
-    assert result2["type"] == "form"
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_form_already_configured(hass, picnic_api):
+    """Test that an entry with unique id can only be added once."""
+    # Create a mocked config entry and make sure to use the same user_id as set for the picnic_api mock response.
+    MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=picnic_api().get_user()["user_id"],
+        data={CONF_ACCESS_TOKEN: "a3p98fsen.a39p3fap", CONF_COUNTRY_CODE: "NL"},
+    ).add_to_hass(hass)
+
+    result_init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    result_configure = await hass.config_entries.flow.async_configure(
+        result_init["flow_id"],
+        {
+            "username": "test-username",
+            "password": "test-password",
+            "country_code": "NL",
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result_configure["type"] == data_entry_flow.RESULT_TYPE_ABORT
```

Comment:
I realize this is an existing issue but -- I think the common way folks do this is to have this check happen in `async_step_user` before moving on to the authenticate step. Can you add a check there for an existing config entry there?

Reply:
You mean aborting if the entry already exists before going into the authenticate step? That's not really possible as we only know the unique id after authentication. It is possible to add the same integration multiple times after all.

Reply:
OK I missed that, thanks for the correction. It may be worth adding test coverage for that case too in a follow up PR unless its already tested and I missed it.